From b516f59490581ff0be4773bf482648fc82f31b3a Mon Sep 17 00:00:00 2001
From: Ryan Houdek <Sonicadvance1@gmail.com>
Date: Tue, 11 Oct 2022 13:04:41 -0700
Subject: [PATCH] vulkan/wsi: Add dep_libudev to idep dependencies

Otherwise users of `idep_vulkan_wsi` won't pull in the udev dependency,
which will cause the linker to fail later on in compiling.

The user of this dependency is lavapipe which would fail to link if this
isn't provided.

Fixes: 4885e63a6d20c57f98b7b641ea3c39a8ff3ae2dd (vulkan/wsi: implement missing wsi_register_device_event)

Reviewed-by: Dylan Baker <dylan@pnwbakers.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/19037>
---
 src/vulkan/wsi/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/vulkan/wsi/meson.build b/src/vulkan/wsi/meson.build
index 75a8025926e..a92c7f3e5eb 100644
--- a/src/vulkan/wsi/meson.build
+++ b/src/vulkan/wsi/meson.build
@@ -88,6 +88,8 @@ else
     # Instruct users of this library to link with --whole-archive.  Otherwise,
     # our weak function overloads may not resolve properly.
     link_whole : libvulkan_wsi,
-    dependencies : idep_vulkan_wsi_headers
+    dependencies : [
+      idep_vulkan_wsi_headers, dep_libudev
+    ]
   )
 endif
